packages/cli/src/cmds/agentInstaller/agentInstallerProcedure.ts

```diff
@@ -56,33 +58,66 @@ export default class AgentInstallerProcedure extends AgentProcedure {
       }
     }
 
-    UI.status = 'Installing the AppMap agent...';
+    UI.status = 'Installing AppMap...';
 
     await this.installer.checkCurrentConfig();
     await this.installer.installAgent();
 
     await this.verifyProject();
 
     const appMapYml = this.configPath;
-    if (!useExistingAppMapYml) {
-      const initCommand = await this.installer.initCommand();
-      const { stdout } = await run(initCommand);
-      const json = JSON.parse(stdout);
+    try {
+      if (!useExistingAppMapYml) {
+        const initCommand = await this.installer.initCommand();
+        const { stdout } = await run(initCommand);
+        const json = JSON.parse(stdout);
 
-      fs.writeFileSync(appMapYml, json.configuration.contents);
-    }
+        fs.writeFileSync(appMapYml, json.configuration.contents);
+      }
+
+      await this.validateProject(useExistingAppMapYml);
 
-    await this.validateProject(useExistingAppMapYml);
+      const successMessage = [
+        chalk.green('Success! AppMap has finished installing.'),
+        '',
+        chalk.blue('NEXT STEP: Record AppMaps'),
+        '',
+        'You can consult the AppMap documentation, or continue with the ',
+        'instructions provided in the AppMap code editor extension.',
+      ];
 
-    const successMessage = [
-      chalk.green('Success! The AppMap agent has been installed.'),
-      '',
-      chalk.blue('NEXT STEP: Record AppMaps'),
-      '',
-      'You can consult the AppMap documentation, or continue with the ',
-      'instructions provided in the AppMap code editor extension.',
-    ];
+      UI.success(successMessage.join('\n'));
+    } catch (e) {
+      const error = e as Error;
+      if (
+        this.installer.name === 'Bundler' &&
+        error?.message.includes('but is an incompatible architecture')
+      ) {
+        await run(new CommandStruct('gem', ['uninstall', 'appmap', '-x', '--force'], this.path));
 
-    UI.success(successMessage.join('\n'));
+        const incompatibleArchitectureMessage = [
+          '\n',
+          chalk.bold.red('AppMap Installation Error!'),
+          '',
+          'Please run the following command in your terminal, then re-run the installer:',
```

Comment:
Removing the gem is a good idea! One small nit-pick: this language looks like it needs to be updated.